cypress-io/cypress
# feat: incorporate v8 snapshot generation into monorepo build process

- Closes https://github.com/cypress-io/cypress/issues/22995
- Closes https://github.com/cypress-io/cypress/issues/22982

### User facing changelog


n/a

### Additional details


This PR does several things:

* Adds a script `tooling/v8-snapshot/dist/setup/setup.js` that can be run in `dev` or `prod` mode that generates a snapshot for the server in the monorepo
* Executes the above script:
  * In the post install script in `dev` mode locally
  * As an additional step of the `build` job in `prod` mode in CI
* Add a hook for `v8-snapshot-require` that will be executed on server startup to hook into the v8 snapshot

Incidental things that were fixed include
* Clean up unused code. The v8 snapshot packages were originally created to be fairly customizable and reusable. I removed the features that we don't immediately need for the sake of long term maintenance

Additionally, there are several existing issues that will be fixed in future iterations. The affected tests were commented out with a note on which future issue would be used to fix them

### Steps to test


You can pull the code locally, run `yarn install` and run `yarn cypress <command>` to verify that things work.

### How has the user experience changed?


n/a

### PR Tasks


- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
